### DISCUSSION TITLE
How can I test it with Jest and React Native Testing Library?

### DISCUSSION THREAD

**Discussion by dutradotdev** [henninghall/react-native-date-picker#423] [Q&A]
Hi,
I want to know how can I test it with Jest and React Native Testing Library? How to mock the DatePicker, select the confirm/cancel button and so on. Do you have a example of it?

**Comment by chika-kasymov** [comment]
@dutradotdev in my case I was able to test it as follows:
```jsx
// in component
<DatePicker
  // other props
  testID="Date Picker"
/>

// in test
const someDate = new Date();
fireEvent(queryByTestId('Date Picker'), 'onChange', {
  nativeEvent: { timestamp: someDate.getTime() },
});
```

**Reply by AbdullahQureshi1080** [reply]
@chika-kasymov Hi, can you please share the tests config file (jest.config), I'm trying to build a component library in which I am trying to test this, but I am having import issues with jest, your config might help.

**Reply by chika-kasymov** [reply]
Hey @AbdullahQureshi1080, here is the config I used:
```javascript
const { defaults: tsjPreset } = require('ts-jest/presets');

module.exports = {
  ...tsjPreset,
  preset: '@testing-library/react-native',
  globals: {
    'ts-jest': {
      tsconfig: 'tsconfig.spec.json',
      babelConfig: true,
      diagnostics: false,
      isolatedModules: true,
    },
  },
  setupFilesAfterEnv: [
    '<rootDir>/test/setup.ts',
    '@testing-library/jest-native/extend-expect',
  ],
  transformIgnorePatterns: [
    'node_modules/(?!(@react-native|@react-native-.*|react-native|react-native-.*|react-navigation|react-navigation-.*|@react-navigation|@react-native-community|@dynatrace|@storybook/)/)',
    '@shared/',
  ],
  testPathIgnorePatterns: [
    '<rootDir>/node_modules/',
    '/patches/',
    '/reports/',
    '/e2e/',
  ],
  testEnvironment: 'node',
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx', 'json', 'node'],
  coveragePathIgnorePatterns: ['.base.ts'],
  moduleNameMapper: {
    '\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$':
      '<rootDir>/assets.transformer.js',
    '^@shared(.*)$': '<rootDir>/shared$1',
  },
};
```

The config contains a couple of lines with the React Native Testing Library settings:
- The **preset** is `'@testing-library/react-native'`.
- The **setupFilesAfterEnv** config contains the `'@testing-library/jest-native/extend-expect'`.

**Reply by shail-spreadd** [reply]
@chika-kasymov queryByTestId is returning as null

**Reply by shail-spreadd** [reply]
now it give me this error
TypeError: Cannot read properties of undefined (reading 'openPicker')

do i have to add mock ?

**Reply by chika-kasymov** [reply]
@shail8266 sorry, I can't help you because I don't work with that project anymore.

**Reply by Ryan-Durling** [reply]
I just ran into this problem that @shail-spreadd brought up

**Reply by shail-spreadd** [reply]
@Ryan-Durling i didn't find any solution so i added istanbul ignore comment for that particular code
if you find any please let me know

**Reply by shail-spreadd** [reply]
@chika-kasymov okay

**Comment by anishsundarjee** [comment]
> Hi, I want to know how can I test it with Jest and React Native Testing Library? How to mock the DatePicker, select the confirm/cancel button and so on. Do you have a example of it?

@dutradotdev I was able to mock this by using jest manual mock [https://jestjs.io/docs/tutorial-react-native#mock-native-modules-using-jestmock](url)

sample code : 
```
  import 'react-native';
  import React from 'react';
  import renderer from 'react-test-renderer';
  import { it, describe } from '@jest/globals';
  import { DateInput } from '@/components';
  jest.mock('react-native-date-picker', () => 'DatePicker');
  
  // run tests as normal
```

Hope this helps
